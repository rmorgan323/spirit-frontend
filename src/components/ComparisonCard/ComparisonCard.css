/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.ComparisonCard {
  background-color: #f1f2f2;
  width: 100px;
  padding: 10px;
  margin: 5px 5px; }
  .ComparisonCard h5 {
    font-size: 14px;
    margin: 0;
    text-align: center;
    height: 35px;
    overflow: scroll;
    word-break: break-word; }
  .ComparisonCard hr {
    border: 0.5px solid #d1d3d4; }
  .ComparisonCard .card-body {
    display: flex;
    margin: auto;
    width: 72px; }
  .ComparisonCard .card-dates {
    margin-right: 10px; }
    .ComparisonCard .card-dates p {
      margin: 0;
      position: relative;
      font-size: 10px;
      color: #a7a9ac;
      line-height: 2.2; }
  .ComparisonCard .card-values p {
    margin: 0;
    font-size: 14px;
    line-height: 1.6; }
