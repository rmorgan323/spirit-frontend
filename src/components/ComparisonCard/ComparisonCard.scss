@import '../../variables.scss';

.ComparisonCard {
  background-color: $grey1;
  width: 100px;
  padding: 10px;
  margin: 5px 5px;


  h5 {
    font-size: 14px;
    margin: 0;
    text-align: center;
    height: 35px;
    overflow: scroll;
    word-break: break-word;
  }

  hr {
    border: .5px solid $grey2;
  }

  .card-body {
    display: flex;
    margin: auto;
    width: 72px;
  }

  .card-dates {
    margin-right: 10px;

    p {
      margin: 0;
      position: relative;
      font-size: 10px;
      color: $grey3;
      line-height: 2.2;
    }
  }

  .card-values {
    p {
      margin: 0;
      font-size: 14px;
      line-height: 1.6;
    }
  }

} 