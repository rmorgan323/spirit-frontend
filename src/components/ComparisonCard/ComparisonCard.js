import React from 'react';
import { PropTypes } from 'prop-types';
import moment from 'moment';
import './ComparisonCard.css';

const ComparisonCard = ({ dateArray, title, values }) => {
  const dates = dateArray.map((date, index) => {
    return <p key={index}>{moment(date).format('MMM D')}</p>;
  });

  const dataValues = values.map((value, index) => {
    if (value === false) {
      value = 'no';
    } else if (value === true) {
      value = 'yes';
    } else if (
      value === null ||
      value === 0 ||
      value === undefined ||
      value === ''
    ) {
      value = '-';
    }

    return <p key={index}>{value}</p>;
  });

  return (
    <div className="ComparisonCard">
      <h5>{title}</h5>
      <hr />
      <div className="card-body">
        <div className="card-dates">
          <p>{dates}</p>
        </div>
        <div className="card-values">
          <p>{dataValues}</p>
        </div>
      </div>
    </div>
  );
};

export default ComparisonCard;

ComparisonCard.propTypes = {
  dateArray: PropTypes.array,
  title: PropTypes.string,
  values: PropTypes.array
};
