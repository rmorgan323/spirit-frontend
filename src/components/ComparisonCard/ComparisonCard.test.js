import React from 'react';
import ComparisonCard from './ComparisonCard';
import { shallow } from 'enzyme';
import {
  mockDataArray,
  mockTitle,
  mockValues
} from '../../data/mockData/mockComparisonCardData';

describe('ComparisonCard tests', () => {
  it('Should match snapshot', () => {
    const renderedApp = shallow(
      <ComparisonCard
        dateArray={mockDataArray}
        title={mockTitle}
        values={mockValues}
      />
    );

    expect(renderedApp).toMatchSnapshot();
  });
});
