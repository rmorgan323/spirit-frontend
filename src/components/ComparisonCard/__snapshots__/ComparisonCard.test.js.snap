// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonCard tests Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    dateArray={
      Array [
        "2018-02-20T20:08:23.713Z",
        "2018-02-20T20:12:14.588Z",
        "2018-02-20T21:25:45.454Z",
      ]
    }
    title="Maladaptive"
    values={
      Array [
        "5r",
        "6i",
        "5f",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5>
          Maladaptive
        </h5>,
        <hr />,
        <div
          className="card-body"
        >
          <div
            className="card-dates"
          >
            <p>
              <p>
                Feb 20
              </p>
              <p>
                Feb 20
              </p>
              <p>
                Feb 20
              </p>
            </p>
          </div>
          <div
            className="card-values"
          >
            <p>
              <p>
                5r
              </p>
              <p>
                6i
              </p>
              <p>
                5f
              </p>
            </p>
          </div>
        </div>,
      ],
      "className": "ComparisonCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Maladaptive",
        },
        "ref": null,
        "rendered": "Maladaptive",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-dates"
            >
              <p>
                <p>
                  Feb 20
                </p>
                <p>
                  Feb 20
                </p>
                <p>
                  Feb 20
                </p>
              </p>
            </div>,
            <div
              className="card-values"
            >
              <p>
                <p>
                  5r
                </p>
                <p>
                  6i
                </p>
                <p>
                  5f
                </p>
              </p>
            </div>,
          ],
          "className": "card-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                <p>
                  Feb 20
                </p>
                <p>
                  Feb 20
                </p>
                <p>
                  Feb 20
                </p>
              </p>,
              "className": "card-dates",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Feb 20
                  </p>,
                  <p>
                    Feb 20
                  </p>,
                  <p>
                    Feb 20
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Feb 20",
                  },
                  "ref": null,
                  "rendered": "Feb 20",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Feb 20",
                  },
                  "ref": null,
                  "rendered": "Feb 20",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Feb 20",
                  },
                  "ref": null,
                  "rendered": "Feb 20",
                  "type": "p",
                },
              ],
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                <p>
                  5r
                </p>
                <p>
                  6i
                </p>
                <p>
                  5f
                </p>
              </p>,
              "className": "card-values",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    5r
                  </p>,
                  <p>
                    6i
                  </p>,
                  <p>
                    5f
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "5r",
                  },
                  "ref": null,
                  "rendered": "5r",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "6i",
                  },
                  "ref": null,
                  "rendered": "6i",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "5f",
                  },
                  "ref": null,
                  "rendered": "5f",
                  "type": "p",
                },
              ],
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5>
            Maladaptive
          </h5>,
          <hr />,
          <div
            className="card-body"
          >
            <div
              className="card-dates"
            >
              <p>
                <p>
                  Feb 20
                </p>
                <p>
                  Feb 20
                </p>
                <p>
                  Feb 20
                </p>
              </p>
            </div>
            <div
              className="card-values"
            >
              <p>
                <p>
                  5r
                </p>
                <p>
                  6i
                </p>
                <p>
                  5f
                </p>
              </p>
            </div>
          </div>,
        ],
        "className": "ComparisonCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Maladaptive",
          },
          "ref": null,
          "rendered": "Maladaptive",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-dates"
              >
                <p>
                  <p>
                    Feb 20
                  </p>
                  <p>
                    Feb 20
                  </p>
                  <p>
                    Feb 20
                  </p>
                </p>
              </div>,
              <div
                className="card-values"
              >
                <p>
                  <p>
                    5r
                  </p>
                  <p>
                    6i
                  </p>
                  <p>
                    5f
                  </p>
                </p>
              </div>,
            ],
            "className": "card-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  <p>
                    Feb 20
                  </p>
                  <p>
                    Feb 20
                  </p>
                  <p>
                    Feb 20
                  </p>
                </p>,
                "className": "card-dates",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Feb 20
                    </p>,
                    <p>
                      Feb 20
                    </p>,
                    <p>
                      Feb 20
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Feb 20",
                    },
                    "ref": null,
                    "rendered": "Feb 20",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "1",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Feb 20",
                    },
                    "ref": null,
                    "rendered": "Feb 20",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "2",
                    "nodeType": "host",
                    "props": Object {
                      "children": "Feb 20",
                    },
                    "ref": null,
                    "rendered": "Feb 20",
                    "type": "p",
                  },
                ],
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  <p>
                    5r
                  </p>
                  <p>
                    6i
                  </p>
                  <p>
                    5f
                  </p>
                </p>,
                "className": "card-values",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      5r
                    </p>,
                    <p>
                      6i
                    </p>,
                    <p>
                      5f
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "host",
                    "props": Object {
                      "children": "5r",
                    },
                    "ref": null,
                    "rendered": "5r",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "1",
                    "nodeType": "host",
                    "props": Object {
                      "children": "6i",
                    },
                    "ref": null,
                    "rendered": "6i",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": "2",
                    "nodeType": "host",
                    "props": Object {
                      "children": "5f",
                    },
                    "ref": null,
                    "rendered": "5f",
                    "type": "p",
                  },
                ],
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
